modules/headers/headers.go

```diff
@@ -0,0 +1,45 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package headers
+
+import (
+	"net/http"
+
+	"github.com/philippta/flyscrape"
+)
+
+func init() {
+	flyscrape.RegisterModule(Module{})
+}
+
+type Module struct {
+	Headers map[string]string `json:"headers"`
+}
+
+func (Module) ModuleInfo() flyscrape.ModuleInfo {
+	return flyscrape.ModuleInfo{
+		ID:  "headers",
+		New: func() flyscrape.Module { return new(Module) },
+	}
+}
+
+func (m Module) AdaptTransport(t http.RoundTripper) http.RoundTripper {
+	if len(m.Headers) == 0 {
+		return t
+	}
+
+	return flyscrape.RoundTripFunc(func(r *http.Request) (*http.Response, error) {
+		h := http.Header{}
+		for k, v := range m.Headers {
+			h.Set(k, v)
+		}
+
+		r.Header = h
```

Comment:
This would override any existing header field on the `http.Request` like those added [here](https://github.com/philippta/flyscrape/blob/47d58e6e0ebc44e7c00dffcc3b892932dc70eb3a/js_lib.go#L85) or [here](https://github.com/philippta/flyscrape/blob/47d58e6e0ebc44e7c00dffcc3b892932dc70eb3a/js_lib.go#L99).

It would be best to set the extra headers directly on the `r.Header` object.

Reply:
i think i need to clarify on this, so we expect to have this custom headers to "add new headers" instead of "replacing the default headers"?

let's have some example:

```
format:
  - default headers
  -- custom headers
  --- final headers

# case 1: empty headers
- {empty headers}
-- "Authorization": "Basic ZGVtbzpwQDU1dzByZA==", "User-Agent": "Gecko/1.0"
--- "Authorization": "Basic ZGVtbzpwQDU1dzByZA==", "User-Agent": "Gecko/1.0"

# case 2: no headers with same key between default and custom
- "User-Agent": "flyscrape/0.1"
-- "Authorization": "Basic ZGVtbzpwQDU1dzByZA=="
--- "Authorization": "Basic ZGVtbzpwQDU1dzByZA==", "User-Agent": "flyscrape/0.1"

# case 3: any headers with same key between default and custom
#   in this case, "User-Agent"
- "User-Agent": "flyscrape/0.1"
-- "Authorization": "Basic ZGVtbzpwQDU1dzByZA==", "User-Agent": "Gecko/1.0"
--- "Authorization": "Basic ZGVtbzpwQDU1dzByZA==", "User-Agent": "Gecko/1.0"
```

LMK if the example doesn't fit the requirement, thanks!

Reply:
Your examples are correct. 

Sorry if that was unclear. The headers module should only ever add or override individual header fields.

Reply:
@philippta no worries, ok i'll update the PR